### PR TITLE
Improve attribute message error spans

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I got the idea while working on https://github.com/rust-lang/rust/pull/79464

### CODE REVIEW

File: compiler/rustc_ast/src/attr/mod.rs [hunk 1]
```diff
@@ -227,6 +240,10 @@ impl MetaItem {
     pub fn is_value_str(&self) -> bool {
         self.value_str().is_some()
     }
+
+    pub fn name_value_literal_span(&self) -> Option<Span> {
```
Comment (oli-obk) [inline compiler/rustc_ast/src/attr/mod.rs:244]: Not sure if this function is worth it

Comment (GuillaumeGomez) [reply compiler/rustc_ast/src/attr/mod.rs:244]: Not sure either. I had the need for it in rustdoc but then removed it. So as you prefer.

Comment (oli-obk) [reply compiler/rustc_ast/src/attr/mod.rs:244]: let's leave it out and just do its body manually at its use sites

Comment (GuillaumeGomez) [reply compiler/rustc_ast/src/attr/mod.rs]: Removed! :)

**Comment by GuillaumeGomez** [comment]
As shown [here](https://github.com/rust-lang/rust/pull/79509/checks?check_run_id=1480797733), it was actually needed. Putting it back then haha.

**Pull request merged**